internal/provider/resource_self_signed_cert_test.go

```diff
@@ -508,3 +531,114 @@ func TestAccResourceSelfSignedCert_FromRSAPrivateKeyResource(t *testing.T) {
 		},
 	})
 }
+
+func TestAccResourceSelfSignedCert_NoSubject(t *testing.T) {
+	r.UnitTest(t, r.TestCase{
+		ProviderFactories: testProviders,
+		Steps: []r.TestStep{
+			{
+				Config: `
+					resource "tls_private_key" "test" {
+						algorithm = "ED25519"
+					}
+					resource "tls_self_signed_cert" "test" {
+						private_key_pem = tls_private_key.test.private_key_pem
+						is_ca_certificate     = true
+						set_subject_key_id    = true
+						validity_period_hours = 8760
+						subject {}
+						allowed_uses = [
+							"key_encipherment",
+							"digital_signature",
+							"server_auth",
+							"client_auth",
+							"cert_signing",
+						]
+						dns_names = [
+							"pippo.pluto.paperino",
+						]
+						ip_addresses = [
+							"127.0.0.2",
+						]
+						uris = [
+							"disney://pippo.pluto.paperino/minnie",
+						]
+					}
+				`,
+				Check: r.ComposeTestCheckFunc(
```

Comment:
Small nits:

- It might be cool to switch this to `ComposeAggregateTestCheckFunc()` similar to other so all check failures are shown at once, rather than only the first.
- Would it be worth having a `testCheckPEMCertificateNoSubject()` helper just to double check things?